pkgs/by-name/go/got/package.nix

```
@@ -37,7 +41,15 @@ stdenv.mkDerivation (finalAttrs: {
     substituteInPlace configure --replace-fail 'xdarwin' 'xhomebrew'
   '';
 
-  env.NIX_CFLAGS_COMPILE = toString (lib.optionals stdenv.isDarwin [
+  env.NIX_CFLAGS_COMPILE = toString (
+  lib.optionals (defaultEditor != null) [
+    ''-DGOT_DEFAULT_EDITOR="${lib.getExe defaultEditor}"''
+  ] ++ lib.optionals withSsh [
+    ''-DGOT_DIAL_PATH_SSH="${lib.getExe openssh}"''
+    ''-DGOT_TAG_PATH_SSH_KEYGEN="${lib.getExe' openssh "ssh-keygen"}"''
+  ] ++ lib.optionals stdenv.isLinux [
+    ''-DGOT_TAG_PATH_SIGNIFY="${lib.getExe signify}"''
+  ] ++ lib.optionals stdenv.isDarwin [
```

```suggestion
  lib.optionals stdenv.cc.isClang [
    "-Wno-error=implicit-function-declaration";
  ] ++ lib.optionals stdenv.isDarwin [
```
seems to also need this based on the CI error? Can't reproduce on Linux though oddly enough

---

I've made the suggested change, @eclairevoyant, for my own interest could you point me to the CI error, please?

---

https://logs.ofborg.org/?key=nixos/nixpkgs.322600&attempt_id=abfa1bf8-2429-45e5-8eef-84abba2145d3

which, looking at it now, I realise might also need `-Wno-error=int-conversion` as well

---

I'd expect these kinds of failures to be also visible here on this PR as check failures. How these do not show up? What am I missing?

---

It was considered a "neutral" check (indicated by a grey square), which doesn't fail the CI but is still reflective of a build failure. I don't know what conditions result in a "neutral" check rather than an explicitly "failed" check.

---

You can see this on https://github.com/NixOS/nixpkgs/commit/75e9a748a25a4723d31aaed9565abf68f73c030a, click on the green check next to the commit message, then scroll down to x86_64-darwin, the grey square is visible there.

---

⬜ doesn't indicate a failure to me as well as a ❌ does. I'll pay closer attention to these in the future, thanks!

And that the checks-menu is ~~hidden~~available from the green check mark is… 🤔… not so ideal. Anyway, I'm ranting 🙂

---

If it was the current commit, you wouldn't need to go to that menu, it'd be in the list of checks below. I only mentioned that menu since the branch is currently pointing to a different commit. But, I agree it's not very obvious.

---

I'll keep a close 👁️ on those checks now 🧐